refarch-cli/index.ts

```diff
@@ -0,0 +1,225 @@
+}
+
+async function generateFrontendInteractiveCli() {
+  const name = await input({
+    message: "Define value for property name:",
+    required: true,
+  });
+  generateFrontend(name);
+}
+
+/**
+ * Generates a new FRONTEND directory with the provided name
+ * The function copies the existing `refarch-frontend` folder to a new folder named `<name>`.
+ * It then performs string replacements in the package.json and package-lock.json to update the name
+ *
+ * @param name - The new name to use for the application
+ */
+function generateFrontend(name: string) {
+  cpSync("refarch-frontend", "refarch-frontend-copy", {
+    recursive: true,
+  });
+  const replacements = {
+    files: [
+      "refarch-frontend-copy/package.json",
+      "refarch-frontend-copy/package-lock.json",
+    ],
+    from: /refarch-frontend/g,
+    to: `package ${name}`,
+  };
+  replaceInFileSync(replacements);
+  renameSync("refarch-frontend-copy", `${name}`);
+}
+
+/**
+ * Generates a new EAI package with the provided package name, groupId and artifactId.
+ * The function copies the existing `refarch-eai` folder to a new folder named `<artifactId>`.
+ * It then performs string replacements on the Java files and pom.xml file within the new folder
+ * to update the package name, groupId, and artifactId.
+ *
+ * @param  packageName - The new package name to use for the Java files.
+ * @param  groupId - The new groupId to use for the pom.xml file.
+ * @param  artifactId - The new artifactId to use for the pom.xml file and the name of the new folder
+ */
+function generateEAI(packageName: string, groupId: string, artifactId: string) {
+  cpSync("refarch-eai", "refarch-eai-copy", { recursive: true });
+  const replacements = [
+    {
+      files: "refarch-eai-copy/src/main/java/de/muenchen/refarch/**/*.java",
+      from: /de.muenchen.refarch/g,
+      to: `${packageName}`,
+    },
+    {
+      files: "refarch-eai-copy/pom.xml",
+      from: [
+        "<groupId>de.muenchen.refarch</groupId>",
+        "<artifactId>refarch-eai</artifactId>",
+        "<name>refarch_eai</name>",
+      ],
+      to: [
+        `<groupId>${groupId}</groupId>`,
+        `<artifactId>${artifactId}</artifactId>`,
+        `<name>${artifactId}</name>`,
+      ],
+    },
+  ];
+  replacements.map((options) => replaceInFileSync(options));
+  renameSync("refarch-eai-copy", `${artifactId}`);
+}
+
+/**
+ *  generated the interactive CLI for choosing, if the Docs and Stack folder should be kept
+ */
+async function generateDefaultCliForDocsAndStack() {
+  keepDocs = await confirm({ message: "Want to keep the Docs folder" });
+  keepStack = await confirm({ message: "Want to keep the Stack folder" });
+  cleanup()
+}
+
+/**
+ * remove the templates for the frontend, backend and eai.
+ * remove the shared files if no java-application got generated
+ * remove the docs and stack depending on the user input
+ */
+function cleanup() {
+  if (!keepDocs) {
+    rmSync("docs", { recursive: true });
```

Comment:
Fails if folder already removed but think that's ok for this version

Reply:
for now, the tool is only for scaffolding newly downloades projects. So yeah its fine for now. More functionality will be added in the future.